presto-main/src/main/java/com/facebook/presto/operator/scalar/RepeatFunction.java

```diff
@@ -0,0 +1,156 @@
+        }
+        for (int i = 0; i < count; i++) {
+            type.writeLong(blockBuilder, element);
+        }
+        return blockBuilder.build();
+    }
+
+    @TypeParameter("T")
+    @SqlType("array(T)")
+    public static Block repeat(
+            @TypeParameter("T") Type type,
+            @SqlNullable @SqlType("T") Slice element,
+            @SqlType(StandardTypes.INTEGER) long count)
+    {
+        BlockBuilder blockBuilder = createBlockBuilder(type, count);
+        if (element == null) {
+            return repeatNullValues(blockBuilder, count);
+        }
+        if (count > 0) {
+            type.writeSlice(blockBuilder, element);
+            checkMaxSize(blockBuilder.getSizeInBytes(), count);
+        }
+        for (int i = 1; i < count; i++) {
+            type.writeSlice(blockBuilder, element);
+        }
+        return blockBuilder.build();
+    }
+
+    @TypeParameter("T")
+    @SqlType("array(T)")
+    public static Block repeat(
+            @TypeParameter("T") Type type,
+            @SqlNullable @SqlType("T") Boolean element,
+            @SqlType(StandardTypes.INTEGER) long count)
+    {
+        BlockBuilder blockBuilder = createBlockBuilder(type, count);
+        if (element == null) {
+            return repeatNullValues(blockBuilder, count);
+        }
+        for (int i = 0; i < count; i++) {
+            type.writeBoolean(blockBuilder, element);
+        }
+        return blockBuilder.build();
+    }
+
+    @TypeParameter("T")
+    @SqlType("array(T)")
+    public static Block repeat(
+            @TypeParameter("T") Type type,
+            @SqlNullable @SqlType("T") Double element,
+            @SqlType(StandardTypes.INTEGER) long count)
+    {
+        BlockBuilder blockBuilder = createBlockBuilder(type, count);
+        if (element == null) {
+            return repeatNullValues(blockBuilder, count);
+        }
+        for (int i = 0; i < count; i++) {
+            type.writeDouble(blockBuilder, element);
+        }
+        return blockBuilder.build();
+    }
+
+    private static BlockBuilder createBlockBuilder(Type type, long count)
+    {
+        checkCondition(count <= MAX_LENGTH, INVALID_FUNCTION_ARGUMENT, "repeat count should be no greater than " + MAX_LENGTH);
+        checkCondition(count >= 0, INVALID_FUNCTION_ARGUMENT, "repeat count must be no smaller than zero");
+        return type.createBlockBuilder(new BlockBuilderStatus(), toIntExact(count));
+    }
+
+    private static Block repeatNullValues(BlockBuilder blockBuilder, long count)
+    {
+        for (int i = 0; i < count; i++) {
+            blockBuilder.appendNull();
+        }
+        return blockBuilder.build();
+    }
+
+    private static void checkMaxSize(long bytes, long count)
+    {
+        checkCondition(bytes <= (MAX_SIZE_IN_BYTES + count) / count,
+                INVALID_FUNCTION_ARGUMENT,
+                "repeat should not create an array greater than %s bytes",
```

Comment:
`result of repeat function must not take more than MAX_SIZE_IN_BYTES bytes`

For coherency, don't use %s.